service/auth/auth.go

```diff
@@ -121,7 +128,7 @@ func WithApiKeySaveOnCreate(saveOnCreate bool) ServiceOption {
 }
 
 // NewService creates a new Service representing an authentication service.
-func NewService(opts ...ServiceOption) *Service {
+func NewService(db persistence.IsDatabase, opts ...ServiceOption) *Service {
```

Comment:
Does it make sense to rather have a service option WithPersistance() and maybe default to an in-memory implementation of it?